### ISSUE TITLE
Blacklisting params does not support nested structures.

### ISSUE THREAD

**Issue by pedelman** [bigcommerce/gruf#22]
In Gruf 1.2.0 parameter blacklisting was added in order to filter out sensitive or long values from logs. 

The current implementation will loop over each of the values in the incoming request object and check if that value is in the blacklist, if so, we will replace with some value (`REDACTED` by default). See: [lib/gruf/instrumentation/request_logging/hook.rb#L126](https://github.com/bigcommerce/gruf/blob/ff818717e8b1db3e03ece521eb0c1c168d673f53/lib/gruf/instrumentation/request_logging/hook.rb#L126)

Say I have a request like the following:

```proto
message UpdateUserReq {
  message Data {
    google.protobuf.StringValue sensitive = 1;
    ...
  }
  uint64 user_id = 1;
  Data data = 2;
}
```

With this structure, the incoming request params will have a shape like the following:

```rb
{:user_id=>263606,
 :data=>
  {:sensitive=>{:value=>"dont log me, or else"}}}
```

If I want to redact `sensitive` from the logs, I would expect to either define `sensitive` in the blacklist configuration, or to specify the key structure so that the blacklist knows how to traverse the hash, like `data.sensitive.value`.

Currently it is not possible to redact fields from within an embedded message.

**Comment by splittingred** [comment]
So I'd really prefer we either:

* Do path matching, such as `data.sensitive.value` or `data.sensitive`, and maintain key structures within
* Do shallow key-based matching, such as `password`, that filters out all keys with the branch key

IMO the blacklist should redact any child keys below a blacklisted value; say here if you had `sensitive` (or `data.sensitive`, it would recursively traverse all keys below and store a `REDACTED` value in every key below it.)

**Comment by splittingred** [comment]
Fixed in #23
